### PR TITLE
Update permissions and regenerate the role

### PR DESCRIPTION
**Pull request by johnbelamaric**
This PR updates the permissions to reflect those that are needed and regenerates the role.

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -83,32 +98,6 @@ rules:
   - get
   - patch
   - update
-- apiGroups:
-  - workload.nephio.org
-  resources:
-  - amfdeployments
```
Comment (johnbelamaric) [inline config/rbac/role.yaml:89]: Oh, this is probably wrong too. Resource names should be all lowercase. Resource names are different than Kind names - they are related but different. Resources relate to the REST calls and get mapped to Kind types. You can theoretically have multiple resources utilizing the same Kinds (though in practice I don't think we do that anywhere in K8s usually).

**Comment by denysaleksandrov** [comment]
@johnbelamaric smf controller creates a service as well, so “service” tag is also required. we missed that when smf pr was merged.
We wanted to change name from free5gc-operator-controller to free5gc-operator. -controller is a bit redundant.
Apart from that - lgtm.

**Comment by johnbelamaric** [comment]
free5gc-operator-controller refers specifically to the role of the controller in the operator. In theory you could have different components that require different roles, that are all part of the operator. But whatever you want to name it is fine with me, I will change it to that.

**Comment by johnbelamaric** [comment]
Ok, done. Adding service to smf didn't make a difference because we only generate one role for all the different controllers in the operator.

**Comment by s3wong** [comment]
> @johnbelamaric smf controller creates a service as well, so “service” tag is also required. we missed that when smf pr was merged. We wanted to change name from free5gc-operator-controller to free5gc-operator. -controller is a bit redundant. Apart from that - lgtm.

@johnbelamaric , @denysaleksandrov  --- actually even AMF creates a "Service" resource as well. But as John said, adding that tag doesn't really make much difference. So LGTM

**Pull request merged**